### ISSUE TITLE
Filter professions

### ISSUE THREAD

**Issue by Nightinggale** [We-the-People-civ4col-mod/Mod#63]
We should make the profession selector able to filter on profession types. We should offer the following lists:
* all (like right now)
* indoor colony professions
* outdoor colony professions (assigned to a plot)
* non-millitary professions
* non-colony professions (pioneers, military etc)

We can also consider writing our own popup in python, in which case we can add multiple scrollbars, one for each kind of profession.

The point is that one very long list is not always able to provide a good overview.

**Comment by Nightinggale** [comment]
Yes that's the popup menu in question. The problem is that later in the game the player can have plenty of professions available in buildings and the number of professions with yield requirements (mainly military) will also be added. In other words the late game menu can be rather long.

I can't remember offhand how the DLL popup menu works regarding popup text, but I know we have complete freedom of the mouseover in the python code. I _think_ we have a decent amount of freedom in the DLL version too. The question is how to access it.

**Comment by abmpicoli** [comment]
Another thing to be thought of : the professions shown at this dialog are only the ones allowed at the tile the player is working with. So,say, changing a farmer that is working at a cleared tile into a lumberjack at the best forest tile for that involves two steps: 
1) assign the farmer to a indoor profession (which is not always possible);
2) once the farmer is at an indoor profession, change the profession to a lumberjack... So we could change the interface to give the option of using the current tile, or the best possible tile.
